benchmarks/ampi/pingpong/pingpong-1way.c

```diff
@@ -1,80 +1,114 @@
 /***********************************************
           MPI / AMPI pingpong test program
           Prints bandwidth and latency for a specific message size
-       
+
           Sameer Kumar 02/08/05
 **************************************/
- 
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <mpi.h>
 
+int finish();
+void run(int msg_size, int iter, int my_id, int p, char *message_s, char *message_r, int printFormat);
+
 int main(int argc, char **argv)
 {
-  int my_id;		/* process id */
-  int p;		/* number of processes */
+  int my_id; /* process id */
+  int p; /* number of processes */
   char* message_s, *message_r;	/* storage for the message */
-  int i, max_msgs, msg_size;
-  MPI_Status status;	/* return status for receive */
-  double elapsed_time_sec;
-  double bandwidth;
-  double startTime = 0;
-  
+  int max_msg_size, min_msg_size, msg_size, low_iter, high_iter, iter, printFormat;
   MPI_Init( &argc, &argv );
-  
+
   MPI_Comm_rank( MPI_COMM_WORLD, &my_id );
   MPI_Comm_size( MPI_COMM_WORLD, &p );
-  
-  if ((sscanf (argv[1], "%d", &max_msgs) < 1) ||
-      (sscanf (argv[2], "%d", &msg_size) < 1)) {
-    fprintf (stderr, "need msg count and msg size as params\n");
-    goto EXIT;
+
+  if(argc < 6) {
+    if(my_id == 0) {
+      printf("Doesn't have required input params. Usage: ./pgm <min-msg-size> <max-msg-size> <low-iter> <high-iter> <print-format (0 for csv, 1 for regular)>\n");
+    }
+    finish();
+  } else {
+    min_msg_size = atoi(argv[1]);
+    max_msg_size  = atoi(argv[2]);
+    low_iter = atoi(argv[3]);
+    high_iter = atoi(argv[4]);
+    printFormat = atoi(argv[5]);
   }
 
-  message_s = (char*)malloc (msg_size);  
-  message_r = (char*)malloc (msg_size);
+  message_s = (char*)malloc (max_msg_size);
+  message_r = (char*)malloc (max_msg_size);
+
+  msg_size = min_msg_size;
+
+  if(my_id < p/2) {
+    if(printFormat == 0)
+      fprintf (stdout, "Msg Size, Iterations, One-way Time (us), Bandwidth (bytes/us)\n");
+    else
+      fprintf (stdout, "%-30s %-25s %-20s %-20s\n", "Msg Size", "Iterations", "One-way Time (us)", "Bandwidth (bytes/us)");
+  }
+
+  while(msg_size <= max_msg_size) {
+    if(msg_size <= 1048576)
+      iter = low_iter;
+    else
+      iter = high_iter;
+    run(msg_size, iter, my_id, p, message_s, message_r, printFormat);
+    msg_size *= 2;
+  }
+
+  free(message_s);
+  free(message_r);
+
+  return finish();
```

Comment:
This seems to return unsuccessful execution (-1) even when it was successful?

Reply:
Removed the `finish` method and added code for successful return.